test/playwright.config.ts

```diff
@@ -12,6 +12,8 @@ const config: PlaywrightTestConfig = {
   testDir: path.join(__dirname, "e2e"), // Search for tests in this directory.
   timeout: 60000, // Each test is given 60 seconds.
   retries: process.env.CI ? 2 : 1, // Retry in CI due to flakiness.
+  // Limit the number of failures on CI to save resources
+  maxFailures: process.env.CI ? 10 : undefined,
```

Comment:
Is 10 too high? Maybe 5?
```suggestion
  maxFailures: process.env.CI ? 5 : undefined,
```

Reply:
Do we know on average how many we usually have?  My sense is it is usually 1-3.

Reply:
Usually it's the state collisions and possibly 1-2 more so I think 3 might be better. Good point